### DISCUSSION TITLE
wgpu::Texture -> wgpu::TextureView ?

### DISCUSSION THREAD

**Discussion by zeroexcuses** [gfx-rs/wgpu#3656] [General]
Repost of https://users.rust-lang.org/t/wgpu-texture-wgpu-textureview/92149

https://docs.rs/wgpu/latest/src/wgpu/lib.rs.html#233-239

I have an `wgpu::Texture`; I want a `wgpu::TextureView`

Now, I see this function `pub fn create_view(&self, desc: &TextureViewDescriptor<'_>) -> TextureView` but I do not have a `TextureViewDescriptor`

Now, `wgpu::Texture` itself has a `TextureViewDescriptor` but it is not public

```
pub struct Texture {
    context: Arc<C>,
    id: ObjectId,
    data: Box<Data>,
    owned: bool,
    descriptor: TextureDescriptor<'static>,
}
```

Is there anyway to do `wgpu::Texture -> wgpu::TextureView` without forking the wgpu package ? I feel like I am missing something very very obvious.

**Comment by cwfitzgerald** [comment]
Hello!

You're slightly misreading the types - a Texture contains it's own descriptor not a view descriptor. 

You can call create_view, TextureViewDescriptor is a fully public struct that you can construct yourself.

**Reply by kpreid** [reply]
Also note that in most cases you don't need to specify any of the fields of the `TextureViewDescriptor`, so a call can be just 

```rust
texture.create_view(&Default::default())
```

The `TextureViewDescriptor` fields are for when you want a view of some _part_ of the texture, like rendering to a specific mip level of a texture, or if you want to provide a `label`.

**Comment by cwfitzgerald** [comment]
Closing as this was also answered on the rust user forum.
